homeassistant/components/matter/helpers.py

```diff
@@ -72,7 +81,11 @@ def node_from_ha_device_id(hass: HomeAssistant, ha_device_id: str) -> MatterNode
     dev_reg = dr.async_get(hass)
     device = dev_reg.async_get(ha_device_id)
     if device is None:
-        raise ValueError("Invalid device ID")
+        raise MissingNode(f"Invalid device ID: {ha_device_id}")
+    config_entry_id = list(device.config_entries)[0]
+    entry = hass.config_entries.async_get_entry(config_entry_id)
+    if entry is None:
+        raise ConfigEntryNotFound(f"Config entry {config_entry_id} not found")
```

Comment:
We also need to make sure that the config entry is loaded, to make it not raise an exception later when trying to get the matter adapter from `hass.data`.

Reply:
I have removed it for now so we unblock this PR (as frontend is waiting on this one) - We'll do a follow-up PR with this addition after having clear what needs to be done.